### PR TITLE
Fixes #11834 - improve host status migration performance

### PR DESCRIPTION
**Pull request by ares**
@ohadlevy could you please test if it helps a bit?

### CODE REVIEW

File: db/migrate/20150612135546_create_host_status.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def up
     add_foreign_key "host_status", "hosts", :name => "host_status_hosts_host_id_fk", :column => 'host_id'
     add_column :hosts, :global_status, :integer, :default => 0, :null => false
 
-    Host.all.each do |host|
+    Host.includes(:host_statuses, :last_report_object).find_each do |host|
```
Comment (dLobatog) [inline db/migrate/20150612135546_create_host_status.rb:13]: `find_in_batches` might be a good one here to improve performance.

Comment (domcleal) [inline db/migrate/20150612135546_create_host_status.rb:13]: http://projects.theforeman.org/issues/11834#note-5 "batch size didn't seem to have much effect"

Comment (ares) [inline db/migrate/20150612135546_create_host_status.rb:13]: `find_in_batches` is used inside `find_each`, it's just a wrapper if you need to work with the whole batch IIUC

**Comment by dLobatog** [comment]
I tested with 17000 hosts:
- Before - https://gist.github.com/dLobatog/0d49460524acdd2bb2f5
- After - https://gist.github.com/96aac4c7205e09595c11 

It seems to strip off around 30s from the migration, but it's still well over 1 minutes. Is this enough @ohadlevy ?

@ares I used this - https://gist.github.com/dLobatog/c5253c189869eb9b9b7d to create the hosts, maybe that can help you when testing it.

**Comment by dLobatog** [comment]
Merged as b2237c5b1dfd83323479d1e586485836cb3b3a20, thanks @ares!

**Pull request closed without merging**